napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,510 @@
+
+    Parameters
+    ----------
+    centers : np.ndarray
+        Mx2 array central coordinates of path triangles.
+    offsets : np.ndarray
+        Mx2 array of the offsets to the central coordinates that need to
+        be scaled by the line width and then added to the centers to
+        generate the actual vertices of the triangulation
+    triangles : np.ndarray
+        (M-2)x3 array of the indices of the vertices that will form the
+        triangles of the triangulation
+    vertex : np.ndarray
+        The vertex of the path for which the centers,
+         offsets and triangles art calculated
+    vec1 : np.ndarray
+        The normal vector from previous vertex to the current vertex
+    vec2 : np.ndarray
+        The normal vector from the current vertex to the next vertex
+    vec1_len : float
+        The length of the vec1 vector, used for miter limit calculation.
+    vec2_len : float
+        The length of the vec2 vector, used for miter limit calculation.
+    j : int
+        The index of position to start putting items in the centers,
+        offsets and triangles arrays
+    cos_limit : float
+        Miter limit which determines when to switch from a miter join to a
+        bevel join
+    bevel : bool
+        Bool which if True causes a bevel join to always be used.
+        If False a bevel join will only be used when the miter limit is exceeded
+
+    Returns
+    -------
+    int
+        number of triangles, centers and offsets added to the arrays
+    """
+    centers[j] = vertex
+    centers[j + 1] = vertex
+    cos_angle = vec1[0] * vec2[0] + vec1[1] * vec2[1]
+    sin_angle = vec1[0] * vec2[1] - vec1[1] * vec2[0]
+    if sin_angle == 0:
+        mitter = np.array([vec1[1], -vec1[0]], dtype=np.float32) * 0.5
+    else:
+        scale_factor = 1 / sin_angle
+        if bevel or cos_limit > cos_angle:
+            # There is a case of bevels join, and
+            # there is a need to check if the miter length is not too long.
+            # For performance reasons here, the mitter length is estimated
+            # by the inverse of the sin of the angle between the two vectors.
+            # See https://github.com/napari/napari/pull/7268#user-content-bevel-cut
+            elapsed_len = scale_factor
+            if vec1_len < vec2_len:
+                if elapsed_len > vec1_len:
+                    scale_factor = vec1_len
+                elif elapsed_len < -vec1_len:
+                    scale_factor = -vec1_len
+            else:
+                if elapsed_len > vec2_len:
+                    scale_factor = vec2_len
+                elif elapsed_len < -vec2_len:
+                    scale_factor = -vec2_len
+
+        # We use here the Intercept theorem for calculating the mitter length
+        # More details in PR description:
+        # https://github.com/napari/napari/pull/7268#user-content-miter
+        mitter = (vec1 - vec2) * 0.5 * scale_factor
```

Comment:
```suggestion
        mitter = (vec1 - vec2) * 0.5 * scale_factor
 
```
Add new line for better readability.